projects/coordinator/src/main/java/org/batfish/coordinator/Main.java

```diff
@@ -183,6 +191,10 @@ private static void initWorkManager() {
 
   }
 
+  public static void initStorage() {
+    _storage = new FileStorageImpl(_settings.getContainersLocation());
```

Comment:
None of the code in a storage consumer should have to know that it's using a FileStorageImpl.

Instead, there should be some sort public factory method `Storage.getImplementation(Settings)` that returns the implementation based on the settings.

Probably, every StorageImpl should have a function like `create(Settings)` that extracts the storage-system-specific settings.